cidc_api/resources/upload_jobs.py

```diff
@@ -145,12 +145,24 @@ def wrapped(args, *pos_args, **kwargs):
 @upload_jobs_bp.route("/<int:upload_job>", methods=["PATCH"])
 @requires_upload_token_auth
 @unmarshal_request(
-    UploadJobSchema(only=["status", "token"]), "upload_job_updates", load_sqla=False
+    UploadJobSchema(only=["status", "gcs_file_map", "token"]),
+    "upload_job_updates",
+    load_sqla=False,
 )
 @marshal_response(upload_job_schema, 200)
-def update_upload_job(upload_job: UploadJobs, upload_job_updates: UploadJobs):
+def update_upload_job(upload_job: UploadJobs, upload_job_updates: dict):
     """Update an upload_job."""
     try:
+        if "gcs_file_map" in upload_job_updates and upload_job.gcs_file_map is not None:
+            upload_job_updates["metadata_patch"] = upload_job.metadata_patch.copy()
+            for uri, uuid in upload_job.gcs_file_map.items():
+                if uri not in upload_job_updates["gcs_file_map"]:
```

Comment:
Do we ever expect a large number of missing optional files? If so, this will run into the same performance issue for merging many artifacts that was solved with the `uuid_path_map` optimization for batch artifact merging here https://github.com/CIMAC-CIDC/cidc-schemas/blob/b74fb7944ef25908796f08ae8e00d4c8fc330a3f/cidc_schemas/prism/merger.py#L139

That said, it's probably better not to prematurely optimize here.

Reply:
yeah, this isn't optimized at all. but hopefully it won't be an issue, otherwise I think we should leave it as "tech debt" for now